lib/helpers/utils.js

```diff
@@ -1770,86 +1782,152 @@ export async function parsePnpmLock(pnpmLock, parentComponent = null) {
     }
     // This logic matches the pnpm list command to include only direct dependencies
     if (ppurl !== "") {
-      // In lock file version 9, direct dependencies is under importers
-      const rootDirectDeps =
-        lockfileVersion >= 9
-          ? yamlObj.importers["."]?.dependencies || {}
-          : yamlObj.dependencies || {};
-      const rootDevDeps =
-        lockfileVersion >= 9
-          ? yamlObj.importers["."]?.devDependencies || {}
-          : {};
-      const rootOptionalDeps =
-        lockfileVersion >= 9
-          ? yamlObj.importers["."]?.optionalDependencies || {}
-          : {};
-      const ddeplist = [];
-      // Find the root optional dependencies
-      for (const rdk of Object.keys(rootDevDeps)) {
-        let version = rootDevDeps[rdk];
-        if (typeof version === "object" && version.version) {
-          version = version.version;
-        }
-        // version: 3.0.1(ajv@8.14.0)
-        if (version?.includes("(")) {
-          version = version.split("(")[0];
-        }
-        const dpurl = new PackageURL(
-          "npm",
-          "",
-          rdk,
-          version,
-          null,
-          null,
-        ).toString();
-        possibleOptionalDeps[decodeURIComponent(dpurl)] = true;
-      }
-      for (const rdk of Object.keys(rootOptionalDeps)) {
-        let version = rootOptionalDeps[rdk];
-        if (typeof version === "object" && version.version) {
-          version = version.version;
-        }
-        // version: 3.0.1(ajv@8.14.0)
-        if (version?.includes("(")) {
-          version = version.split("(")[0];
+      if (yamlObj["importers"]) {
```

Comment:
Can we remove this if block by combining with the above line or using `continue` or `return`?

Reply:
I have added the block above because, when testing with a pnpm file without `importers` the previous statements resulted in error.
I had previously tried to reduce the scope [comment](https://github.com/CycloneDX/cdxgen/pull/1377#discussion_r1759976329)